<<<<<<<< HEAD:package/python3/0013-Add-an-option-to-disable-the-curses-module.patch
From a295784d19553cc28e4d531c0f21ac2935b8bef9 Mon Sep 17 00:00:00 2001
========
From d843bceff915b2a50b36a9db8552e0db5bd90f6d Mon Sep 17 00:00:00 2001
>>>>>>>> upstream/master:package/python3/0012-Add-an-option-to-disable-the-curses-module.patch
From: Thomas Petazzoni <thomas.petazzoni@free-electrons.com>
Date: Wed, 22 Feb 2017 17:31:51 -0800
Subject: [PATCH] Add an option to disable the curses module

Signed-off-by: Thomas Petazzoni <thomas.petazzoni@free-electrons.com>
Signed-off-by: Samuel Martin <s.martin49@gmail.com>
[ Andrey Smirnov: ported to Python 3.6 ]
Signed-off-by: Andrey Smirnov <andrew.smirnov@gmail.com>
[ Adam Duskett: ported to Python 3.10.0 ]
Signed-off-by: Adam Duskett <aduskett@gmail.com>
---
 Makefile.pre.in | 5 ++++-
 configure.ac    | 7 +++++++
 2 files changed, 11 insertions(+), 1 deletion(-)

diff --git a/Makefile.pre.in b/Makefile.pre.in
<<<<<<<< HEAD:package/python3/0013-Add-an-option-to-disable-the-curses-module.patch
index 8af8a3aa40b..7162d9d3634 100644
--- a/Makefile.pre.in
+++ b/Makefile.pre.in
@@ -2093,7 +2093,6 @@ LIBSUBDIRS=	asyncio \
========
index 6ab184fe5f8..09408c3471f 100644
--- a/Makefile.pre.in
+++ b/Makefile.pre.in
@@ -1918,7 +1918,6 @@ LIBSUBDIRS=	asyncio \
>>>>>>>> upstream/master:package/python3/0012-Add-an-option-to-disable-the-curses-module.patch
 		concurrent concurrent/futures \
 		csv \
 		ctypes ctypes/macholib \
-		curses \
 		dbm \
 		email email/mime \
<<<<<<<< HEAD:package/python3/0013-Add-an-option-to-disable-the-curses-module.patch
 		encodings \
@@ -2246,6 +2245,10 @@ LIBSUBDIRS += tkinter
 TESTSUBDIRS += test/test_tkinter test/test_ttk
========
@@ -2084,6 +2083,9 @@ TESTSUBDIRS += tkinter/test tkinter/test/test_tkinter \
 	tkinter/test/test_ttk
>>>>>>>> upstream/master:package/python3/0012-Add-an-option-to-disable-the-curses-module.patch
 endif
 
+ifeq (@CURSES@,yes)
+LIBSUBDIRS += curses
+endif
+
 COMPILEALL_OPTS=-j0
 
 ifeq (@PYDOC@,yes)
diff --git a/configure.ac b/configure.ac
<<<<<<<< HEAD:package/python3/0013-Add-an-option-to-disable-the-curses-module.patch
index 7caa361138d..748a16d7eab 100644
--- a/configure.ac
+++ b/configure.ac
@@ -4497,6 +4497,13 @@ AC_ARG_ENABLE(sqlite3,
 AS_IF([test "$SQLITE3" = "no"],
       [PY_STDLIB_MOD_SET_NA([_sqlite3])])
========
index 342b295bcdd..5d52c813ba6 100644
--- a/configure.ac
+++ b/configure.ac
@@ -4240,6 +4240,15 @@ if test "$TK" = "no"; then
    DISABLED_EXTENSIONS="${DISABLED_EXTENSIONS} _tkinter"
 fi
>>>>>>>> upstream/master:package/python3/0012-Add-an-option-to-disable-the-curses-module.patch
 
+AC_SUBST(CURSES)
+AC_ARG_ENABLE(curses,
+	AS_HELP_STRING([--disable-curses], [disable curses]),
+	[ CURSES="${enableval}" ], [ CURSES=yes ])
+AS_IF([test "$CURSES" = "no"],
+      [PY_STDLIB_MOD_SET_NA([_curses], [_curses_panel])])
+
 AC_SUBST(PYDOC)
 
 AC_ARG_ENABLE(pydoc,
-- 
<<<<<<<< HEAD:package/python3/0013-Add-an-option-to-disable-the-curses-module.patch
2.43.0
========
2.44.0
>>>>>>>> upstream/master:package/python3/0012-Add-an-option-to-disable-the-curses-module.patch

